CMakeLists.txt

```diff
@@ -4,13 +4,9 @@ project(dmclock CXX)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake/modules")
 
-if (NOT(BOOST_FOUND))
-  find_package(Boost REQUIRED)
-endif()
-
 find_package(Threads)
 
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
this setting is introduced in cmake 3.8, but we need to support cmake 3.5.1, please refer to  

https://github.com/ceph/ceph/blob/5a3344f0e52c05c5f14c326e676bb9e5a065b82b/src/CMakeLists.txt#L102-L125

on how to appease cmake 3.5.1